app/controllers/journals_controller.rb

```diff
@@ -98,7 +98,7 @@ def valid_field?(field)
 
   def valid_diff?
     valid_field?(params[:field]) &&
-      @journal.journable.instance_of?(WorkPackage)
```

Comment:
@cbliard do you foresee any problems with this? It was the easiest way to get diff to play nice for project activity imo

Reply:
@mrmir This condition seems strange as it  restricts the usage of diffing things to `WorkPackage`, and thus reduce reusability.
I inspected the history of the file with `git log ` to find why this condition has been introduced.
It has been added in cc2afee99c. The commit message states this:

>    fix controller response
>   
>   As the diff action claims to be more generic, react with a 404 instead
>   of a 403 whenever the journable type is not a WorkPackage.
>   
>   This makes the the use of the diff action less generic, as a user will
>   still need the view_work_packages permission to access the diff action.

That's interesting! It looks like this endpoint requires the `view_work_packages` permission to work. So I tried disabling the work package tracking module and it successfully lead to an error: in this configuration, displaying the diff of a project description change will lead to a 403.

This should be fixed as it should work even if the work package tracking module is disabled.

Reply:
I handled it in 3c4742792b.